src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -42,8 +46,7 @@
 class Index:
     def __init__(
         self,
-        # TODO: SNOW-1481037 : Fix typehints for index constructor, set_query_compiler and set_local_index
-        data: ArrayLike | Any = None,
+        data: ArrayLike | SnowflakeQueryCompiler = None,
```

Comment:
```suggestion
        data: ArrayLike | SnowflakeQueryCompiler | None = None,
```
You should add the extra `None` annotation to make clear this is optional.